### PR TITLE
chore: fix regex on components and oneOf replace

### PR DESCRIPTION
**Pull request by kleyow**

### CODE REVIEW

File: fspiop/v2_0_ISO20022/openapi3/components/schemas/BaseOneRate.yaml [hunk 1]
```diff
@@ -2,5 +2,4 @@ title: BaseOneRate
 type: string
 description: |
   Rate expressed as a decimal, for example, 0.7 is 7/10 and 70%.
-pattern: ^(?=[\d\-\.\,]{1,11}$)([\d\-\,\.]{1,10})?$
```
Comment (PaulGregoryBaker) [inline fspiop/v2_0_ISO20022/openapi3/components/schemas/BaseOneRate.yaml:5]: If the json schema for this is:
  <xs:simpleType name="BaseOneRate">
        <xs:restriction base="xs:decimal">
            <xs:fractionDigits value="10"/>
            <xs:totalDigits value="11"/>
        </xs:restriction>
    </xs:simpleType>
The this regex is all the possible options.    
pattern: ^(?!0\d)(\d{1,11}|\d{1,10}\.\d{1}|\d{1,9}\.\d{1,2}|\d{1,8}\.\d{1,3}|\d{1,7}\.\d{1,4}|\d{1,6}\.\d{1,5}|\d{1,5}\.\d{1,6}|\d{1,4}\.\d{1,7}|\d{1,3}\.\d{1,8}|\d{1,2}\.\d{1,9}|\d{1}\.\d{1,10})$

That seems a bit excessive. Maybe this is ok. I.e. is only support max 6 orders of magnitiude difference is currencies. 
pattern: ^(?!0\d)(\d{1,6}|\d{1,5}\.\d{1,5}|\d{1,4}\.\d{1,6}|\d{1,3}\.\d{1,7}|\d{1,2}\.\d{1,8}|\d{1}\.\d{1,9})$

File: fspiop/v2_0_ISO20022/openapi3/components/schemas/IBAN2007Identifier.yaml [hunk 2]
```diff
@@ -8,5 +8,4 @@ description: >
   International Bank Account Number (IBAN)" and replaced by the more recent
   edition of the standard.
 type: string
-pattern: ^[A-Z]{2,2}[0-9]{2,2}[a-zA-Z0-9]{1,30}$
```
Comment (PaulGregoryBaker) [inline fspiop/v2_0_ISO20022/openapi3/components/schemas/IBAN2007Identifier.yaml:11]: This pattern is also used in the json scheme. Not sure why it need to be deleted.

File: fspiop/v2_0_ISO20022/openapi3/components/schemas/Number.yaml [hunk 3]
```diff
@@ -1,4 +1,3 @@
 title: Number
 type: string
-pattern: ^[\d\-\.\,]{1,18}$
```
Comment (PaulGregoryBaker) [inline fspiop/v2_0_ISO20022/openapi3/components/schemas/Number.yaml:3]: ^(?!0\d)\d{1,18}$

I don't think we should support the negative.
Not sure why it is deleted.

File: fspiop/v2_0_ISO20022/openapi3/components/schemas/PercentageRate.yaml [hunk 4]
```diff
@@ -1,4 +1,3 @@
 title: PercentageRate
 type: string
-pattern: ^(?=[\d\-\.\,]{1,11}$)([\d\-\,\.]{1,10})?$
```
Comment (PaulGregoryBaker) [inline fspiop/v2_0_ISO20022/openapi3/components/schemas/PercentageRate.yaml:3]: ^(?!0\d)(\d{1,11}|\d{1,10}\.\d{1}|\d{1,9}\.\d{1,2}|\d{1,8}\.\d{1,3}|\d{1,7}\.\d{1,4}|\d{1,6}\.\d{1,5}|\d{1,5}\.\d{1,6}|\d{1,4}\.\d{1,7}|\d{1,3}\.\d{1,8}|\d{1,2}\.\d{1,9}|\d{1}\.\d{1,10})$

File: fspiop/v2_0_ISO20022/openapi3/components/schemas/hexBinary.yaml [hunk 5]
```diff
@@ -1,4 +1,3 @@
 title: hexBinary
 type: string
-pattern: ^(0x|0X)?[a-fA-F0-9]+$
```
Comment (PaulGregoryBaker) [inline fspiop/v2_0_ISO20022/openapi3/components/schemas/hexBinary.yaml:3]: Maybe we should align with the FSPIOP ILP packet regex.
I.e. ^[A-Za-z0-9-_]+[=]{0,2}$

Not sure why _ and == are allowed, but I am sure there it a good reason.

**Review by PaulGregoryBaker — COMMENTED** [review]
Added some comments.

**Review by PaulGregoryBaker — APPROVED** [review]
Looks good.

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**